app/models/attendance.rb

```diff
@@ -22,6 +22,7 @@ def take_participant_attendance
       student = meeting.find_student_from_participant(participant)
       next if student.nil?
       student_attendance = student_attendances.find_or_create_by(student: student)
+      student_attendance.update(duration: participant.duration) if participant.class == ZoomParticipant
```

Comment:
Let's take out the if statement here and do an update on all student_attendances regardless of whether they were on slack or zoom. You can have the SlackThread return a duration of `nil` and then we can update it later using the slack_presence_checks data.